front-end/js/pageOnLoad.js

```diff
@@ -134,6 +144,10 @@ function pageOnLoad(serverAddr) {
             );
             processRenderingParams();
 
+            // process user-defined CSS styles
+            globalVar.styles = globalVar.project.styles;
+            processStyles();
+
```

Comment:
minor: by the time /first returns, `globalVar.project.styles` is ready. So you can directly access it in `processStyles()`, rather than pass it to another variable `globalVar.styles`